File: backend/app/models/campaign.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import Column, String, DateTime, Text
from sqlalchemy.dialects.postgresql import UUID, JSON

from app.db.base import Base


class Campaign(Base):
    __tablename__ = "campaigns"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)

    tenant_id = Column(UUID(as_uuid=True), nullable=False, index=True)

    # 🔗 LINK TO PRODUCT + TEMPLATE (SUNGURA CORE)
    product_id = Column(UUID(as_uuid=True), nullable=False)  # ✅ KEEP (single product support)
    template_id = Column(String, nullable=False)

    # ✅ NEW: MULTI-PRODUCT SUPPORT (MVP SAFE)
    product_ids = Column(JSON, nullable=True)

    # OPTIONAL HUMAN LABEL
    name = Column(String, nullable=True)

    # MULTI-PLATFORM SUPPORT (FUTURE SAFE)
    platforms = Column(JSON, nullable=False, default=["facebook"])

    # MULTI-PAGE SUPPORT
    page_ids = Column(JSON, nullable=False)

    # CONTENT
    caption = Column(Text, nullable=False)

    # ✅ SINGLE MEDIA (existing)
    media_url = Column(Text, nullable=True)

    # ✅ NEW: MULTI-MEDIA (scrollable posts)
    media_urls = Column(JSON, nullable=True)

    # SCHEDULING (UTC AWARE)
    scheduled_at = Column(DateTime(timezone=True), nullable=True)

    # EXECUTION STATE
    status = Column(String, default="draft")
    # draft | scheduled | processing | posted | failed

    # TIMESTAMPS
    created_at = Column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc)
    )

    updated_at = Column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc)
    )

File: frontend/src/types/campaign.ts
// types/campaign.ts — POSTIKA
// Extended from original type — backward compatible.
// Added: product_ids, media_urls, created_at, updated_at, template_id

export type Campaign = {
  id: string;
  caption: string;
  // Original single media field — kept for backward compat
  media_url: string | null;
  // Multi-product media list
  media_urls?: string[];
  status: "scheduled" | "processing" | "posted" | "failed" | "draft";
  scheduled_at: string | null;
  created_at?: string;
  updated_at?: string;
  platforms: string[];
  page_ids: string[];
  // Multi-product IDs (replaces single product_id)
  product_ids?: string[];
  template_id?: string;
};

export type PostHistory = {
  id: string;
  status: string;
  // Original field — kept as-is
  external_post_id: string | null;
  created_at: string;
};

// ── Request types ─────────────────────────────────────────────────────────────

export type CreateCampaignRequest = {
  caption: string;
  // product_ids[] — not product_id
  product_ids: string[];
  template_id?: string;
  page_ids: string[];
  platforms: string[];
  // Multi-image
  media_urls?: string[];
  // Legacy fallback
  media_url?: string;
  scheduled_at?: string;
};

export type UpdateCampaignRequest = Partial<CreateCampaignRequest>;

// ── AI generation ─────────────────────────────────────────────────────────────

export type AIGenerateRequest = {
  // product_ids[] for multi-product AI generation
  product_ids: string[];
  template_id?: string;
};

export type AIGenerateResponse = {
  full_caption?: string;
  caption?: string;
  text?: string;
  content?: string;
  result?: string;
};